qiskit/chemistry/ground_state_calculation/ground_state_calculation.py

```diff
@@ -0,0 +1,64 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The ground state calculation interface."""
+
+from abc import ABC, abstractmethod
+
+from qiskit.chemistry.drivers import BaseDriver
+from qiskit.chemistry.results import EigenstateResult
+
+from ..qubit_transformations.qubit_operator_transformation import QubitOperatorTransformation
+
+
+class GroundStateCalculation(ABC):
```

Comment:
This mixes base class implementation with an interface. I think an interface for a ground state calculation is just the `compute_groundstate` and `returns_groundstate` which could be split out and then this class implements that too and acts as a base class for the current set of modules we have and does not preclede other ground state calculations from just implementing that interface that do not want to extend this class.

Also can can we do some better naming on the methods - returns_groundstate being False seems like compute_groundstate should not return a groundstate since their names are so similar - how about something like has_(eigen)value_only, or even returns_state and drop the ground part to decouple them a bit

Reply:
I think a "proper" interface with only `compute_groundstate` and `returns_groundstate` makes sense when we have a use case. For now, since all GSC use a transformation, I think it's fine like this -- once we have an algo that does not use it we can still add an interface underneath w/o users noticing it. 🙂 

That's right, these names might be confusing -- it's on the list for possible renaming.

Reply:
> once we have an algo that does not use it we can still add an interface underneath w/o users noticing it

Its also for users that might want to build one themselves, not just for us. But then you know my opinion of mixing interface and base class implementation from AE as well.